iohk/UpdateLogic.hs

```diff
@@ -106,23 +142,27 @@ data VersionJson = VersionJson {
 
 instance ToJSON VersionJson
 
-extractVersionFromTravis :: LBS.ByteString -> Maybe T.Text
-extractVersionFromTravis yaml = do
-  let
-    f1 = parseTravisYaml yaml
-    f2 :: Either String TravisYaml -> TravisYaml
-    f2 (Right v)  = v
-    f2 (Left err) = error $ "unable to parse .travis.yml: " <> err
-    f3 = global $ env (f2 f1)
-  lookup' "VERSION" f3
+extractVersionFromTravis :: TravisYaml -> Maybe T.Text
+extractVersionFromTravis yaml = lookup' "VERSION" (global (env yaml))
+
+-- | Get VERSION environment variable from pipeline definition.
+-- First looks in global env, otherwise finds first build step with a version.
+extractVersionFromBuildkite :: BK.PipelineDefinition -> Maybe T.Text
+extractVersionFromBuildkite BK.PipelineDefinition{..} = getVer plEnv <|> headMay stepVers
```

Comment:
Most of these parsing functions would go away if we stored versions in a central place in git repos.

We can leave it as is, but once DEVOPS-59 is done this won't be needed anymore.